packages/account-usage/lib/events/onUsageIncreased.ts

```diff
@@ -0,0 +1,88 @@
+import db from '@nangohq/database';
+import { getLocking } from '@nangohq/kvstore';
+import { accountService, getPlan, userService } from '@nangohq/shared';
+import { report, stringifyError } from '@nangohq/utils';
+
+import { sendUsageLimitReachedEmail, sendUsageNearLimitEmail } from '../emails.js';
+import { getAccountUsageTracker } from '../index.js';
+
+import type { AccountUsageMetric } from '../index.js';
+import type { DBPlan } from '@nangohq/types';
+
+export async function onUsageIncreased({
+    accountId,
+    metric,
+    delta,
+    plan
+}: {
+    accountId: number;
+    metric: AccountUsageMetric;
+    delta: number;
+    plan?: DBPlan | undefined;
+}) {
+    try {
```

Comment:
can we had tracing (a single span for the entire function is ok for now) so we can keep an eye on the perf